File: sem4/mpp/mpp-proiect-java-vreauladudu/src/main/java/moto/repository/IRepository.java
package moto.repository;

import moto.domain.Entity;

import java.util.Optional;

public interface IRepository<ID, E extends Entity<ID>>
{
    Optional<E> find(ID id);
    Iterable<E> findAll();
    Optional<E> save(E entity);
    Optional<E> remove(ID id);
    Optional<E> update(E entity);
}

File: sem4/mpp/mpp-proiect-csharp-vreauladudu/repository/IRepository.cs
using mpp_proiect_csharp_vreauladudu.domain;

namespace mpp_proiect_csharp_vreauladudu.repository
{
    internal interface IRepository<ID, E> where E : Entity<ID>
    {
        E? find(ID id);
        IEnumerable<E> findAll();
        E? save(E e);
        E? remove(ID id);
        E? update(E e);
    }
}